File: m4d/ViewModels/ActivityLogPageModel.cs
namespace m4d.ViewModels;

/// <summary>
/// Safe-to-serialize projection of a single ActivityLog entry.
/// </summary>
public class ActivityLogEntry
{
    public int Id { get; set; }
    public DateTimeOffset Date { get; set; }
    public string UserName { get; set; }
    public string Action { get; set; }
    public string Details { get; set; }
}

/// <summary>
/// Top-level model for the activity log Vue page.
/// Server handles paging; Vue renders the current page and builds navigation URLs.
/// </summary>
public class ActivityLogPageModel
{
    public List<ActivityLogEntry> Entries { get; set; }
    public int Page { get; set; }
    public int TotalPages { get; set; }
}

File: m4d/ClientApp/src/pages/activity-log/ActivityLogPageModel.ts
import { jsonArrayMember, jsonMember, jsonObject } from "typedjson";

@jsonObject
export class ActivityLogEntry {
  @jsonMember(Number) public id!: number;
  @jsonMember(String) public date!: string;
  @jsonMember(String) public userName?: string;
  @jsonMember(String) public action!: string;
  @jsonMember(String) public details?: string;
}

@jsonObject
export class ActivityLogPageModel {
  @jsonArrayMember(ActivityLogEntry) public entries!: ActivityLogEntry[];
  @jsonMember(Number) public page!: number;
  @jsonMember(Number) public totalPages!: number;
}
